pkg/controller/storagecluster/storagecluster.go

```diff
@@ -334,6 +339,78 @@ func (c *Controller) validateCustomAnnotations(current *corev1.StorageCluster) e
 	return nil
 }
 
+
+
+func getKeysFromNodeSelector(ns *corev1.NodeSelector) map[string]bool {
+	keys := make(map[string]bool)
+
+	// Use node name as selector.
+	if ns.NodeName != "" {
+		keys["NodeName"] = true
+	}
```

Comment:
If cloud storage is not empty, then we should not even allow `nodeName` selector. The problem would be that if different nodes define different cloud storage configurations, then we won't be able to honor them on restarts. Same as different label selector key problem we are trying to solve/validate.

Reply:
"different nodes define different cloud storage configurations, then we won't be able to honor them on restarts", the node name does not change upon restarts and the cloud storage configuration should be honored right?

Reply:
The issue here is how clouddrive is implemented. Currently the entire logic is in the px layer. The operator sends the params to the px container, but when failover of clouddrive happens in the background, the original spec may not be honored. This could be confusing to the user. That is why we are adding these validations. We can talk offline if you need more context.